tests/ServerTest.php

```diff
@@ -225,6 +225,29 @@ public function testRequestConnectAuthorityForm()
         $this->assertSame('example.com:443', $requestAssertion->getHeaderLine('Host'));
     }
 
+    public function testRequestConnectWithoutHostWillBeAdded()
+    {
+        $requestAssertion = null;
+        $server = new Server(function (ServerRequestInterface $request) use (&$requestAssertion) {
+            $requestAssertion = $request;
+            return new Response();
+        });
+
+        $server->listen($this->socket);
+        $this->socket->emit('connection', array($this->connection));
+
+        $data = "CONNECT example.com:443 HTTP/1.1\r\n\r\n";
+        $this->connection->emit('data', array($data));
+
+        $this->assertInstanceOf('RingCentral\Psr7\Request', $requestAssertion);
```

Comment:
Shouldn't this test against `Psr\Http\Message\ServerRequestInterface` instead of the implementation?